### PR TITLE
timer: Select APIC TSC as a default x86 timer and refactor selection

### PR DESCRIPTION
**Pull request by finikorg**
Allow to select only one timer and choose default timer `APIC_TSC_DEADLINE_TIMER`.

### CODE REVIEW

File: drivers/timer/Kconfig.x86 [hunk 1]
```diff
@@ -1,12 +1,30 @@
...
+	select LOAPIC
+	imply TIMER_READS_ITS_FREQUENCY_AT_RUNTIME
+	select TICKLESS_CAPABLE
+	select TIMER_HAS_64BIT_CYCLE_COUNTER
+	help
+	  This option selects High Precision Event Timer (HPET) as a
+	  system timer.
+
+config APIC_TIMER
+	bool "New local APIC timer"
```
Comment (dcpleung) [inline drivers/timer/Kconfig.x86:28]: Maybe we should just skip the "New" part. :)

File: drivers/timer/Kconfig.x86 [hunk 2]
```diff
@@ -16,6 +34,24 @@ menuconfig APIC_TIMER
 	  without complete APIC emulation).  Modern hardware will work
 	  better with CONFIG_APIC_TSC_DEADLINE_TIMER.
 
+config APIC_TSC_DEADLINE_TIMER
+	bool "Even newer APIC timer using TSC deadline mode"
```
Comment (dcpleung) [inline drivers/timer/Kconfig.x86:40]: Ditto.

File: drivers/timer/Kconfig.x86 [hunk 3]
```diff
@@ -18,6 +18,7 @@ config HPET_TIMER
 	imply TIMER_READS_ITS_FREQUENCY_AT_RUNTIME
 	select TICKLESS_CAPABLE
 	select TIMER_HAS_64BIT_CYCLE_COUNTER
+	select DEPRECATED
```
Comment (dcpleung) [inline drivers/timer/Kconfig.x86]: Hm... I don't think HPET should be deprecated. It can still be used.

File: drivers/timer/Kconfig.x86 [hunk 4]
```diff
@@ -5,6 +5,7 @@
 
 choice
 	prompt "Default System Timer"
+	default HPET_TIMER if SOC_FAMILY_INTEL_ISH || SOC_IA32 || SOC_LAKEMONT
```
Comment (dcpleung) [inline drivers/timer/Kconfig.x86:8]: ... hence I think HPET should not be marked deprecated.

Comment (finikorg) [reply drivers/timer/Kconfig.x86:8]: Removed deprecation of HPET

**Review by dcpleung — APPROVED** [review]

**Review by teburd — APPROVED** [review]

**Review by kwd-doodling — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**